### PR TITLE
[UPD] upgrade this script(compare_noupdate_xml_records.py) to be compatible with 10.0 and small bugs of module: openupgrade_records

### PR DESCRIPTION
**Pull request by duanyp1991**
Changes to be committed:
	modified:   scripts/compare_noupdate_xml_records.py

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: scripts/compare_noupdate_xml_records.py [hunk 1]
```diff
@@ -105,8 +110,20 @@ def process_data_node(data_node):
                 tree = etree.parse(os.path.join(addon_dir, *xml_path))
             except etree.XMLSyntaxError:
                 continue
-            for data_node in tree.xpath("/openerp/data"):
-                process_data_node(data_node)
+            # FIX by Alex: Make it compatible between the 9.0 and 10.0 version;
```
Comment (pedrobaeza) [inline scripts/compare_noupdate_xml_records.py]: Don't put text "Fix by Alex". This is already got by commit author.

File: scripts/compare_noupdate_xml_records.py [hunk 2]
```diff
@@ -105,8 +110,20 @@ def process_data_node(data_node):
...
+            if version == 'old':
+                for data_node in tree.xpath("/openerp/data"):
+                    process_data_node(data_node)
+            elif version == 'new':
```
Comment (pedrobaeza) [inline scripts/compare_noupdate_xml_records.py]: This doesn't totally serve, as you can have in 9.0 both `<odoo>` or `<openerp>` tags.

**Comment by pedrobaeza** [comment]
Can you please attend our comments?

File: odoo/addons/openupgrade_records/models/comparison_config.py [hunk 3]
```diff
@@ -50,8 +50,8 @@ def test_connection(self):
             raise UserError(
                 _("Connection failed."), unicode(e))
         raise UserError(
-            _("Connection succesful."),
-            _("%s is connected.") % user_info["name"])
+            _("Connection succesful."
+                "%s is connected.") % user_info["name"])
```
Comment (pedrobaeza) [inline odoo/addons/openupgrade_records/models/comparison_config.py]: bad indentation

File: scripts/compare_noupdate_xml_records.py [hunk 4]
```diff
@@ -63,7 +63,7 @@ def update_node(target, source):
         target.append(element)
 
 
-def get_records(addon_dir):
+def get_records(addon_dir, version="old"):
```
Comment (pedrobaeza) [inline scripts/compare_noupdate_xml_records.py]: No need for the new argument

File: scripts/compare_noupdate_xml_records.py [hunk 5]
```diff
@@ -140,8 +155,8 @@ def main(argv=None):
         'newdir', metavar='newer_module_directory')
     arguments = parser.parse_args(argv)
 
-    old_update, old_noupdate = get_records(arguments.olddir)
-    new_update, new_noupdate = get_records(arguments.newdir)
+    old_update, old_noupdate = get_records(arguments.olddir, version="old")
```
Comment (pedrobaeza) [inline scripts/compare_noupdate_xml_records.py]: Not needed to call this 2 times

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: odoo/addons/openupgrade_records/models/comparison_config.py [hunk 6]
```diff
@@ -50,8 +50,7 @@ def test_connection(self):
             raise UserError(
                 _("Connection failed."), unicode(e))
```
Comment (ateijelo) [inline odoo/addons/openupgrade_records/models/comparison_config.py]: This UserError call needs to be fixed as well

**Comment by StefanRijnhart** [comment]
@duanyp1991 did you send your CLA to the OCA? We would very much like to take your contribution!

File: odoo/addons/openupgrade_records/models/comparison_config.py (hunk 6 above)
Comment (duanyp1991) [reply odoo/addons/openupgrade_records/models/comparison_config.py]: fixed.

File: odoo/addons/openupgrade_records/models/comparison_config.py [hunk 7]
```diff
@@ -47,11 +47,9 @@ def test_connection(self):
             ids = user_model.search([("login", "=", "admin")])
             user_info = user_model.read(ids[0], ["name"])
         except Exception, e:
-            raise UserError(
-                _("Connection failed."), unicode(e))
+            raise UserError(u'%s%s' % (_("Connection failed.\n\nDETAIL: "), str(e)))
```
Comment (StefanRijnhart) [inline odoo/addons/openupgrade_records/models/comparison_config.py]: How about
```
raise UserError(_("Connection failed.\n\nDETAIL: %s") % unicode(e))
```
Would that work?

**Comment by StefanRijnhart** [comment]
I got notified of your CLA. Thank you!

File: odoo/addons/openupgrade_records/models/comparison_config.py (hunk 7 above)
Comment (duanyp1991) [reply odoo/addons/openupgrade_records/models/comparison_config.py]: your way will work but not as friendly to odoo translation I think .

File: scripts/compare_noupdate_xml_records.py [hunk 8]
```diff
@@ -185,12 +200,13 @@ def main(argv=None):
         if len(element):
             data.append(element)
 
-    openerp = etree.Element("openerp")
-    openerp.append(data)
-    document = etree.ElementTree(openerp)
+    odoo = etree.Element("odoo")
+    odoo.append(data)
```
Comment (StefanRijnhart) [inline scripts/compare_noupdate_xml_records.py]: You can leave out the data level here, and produce files like
```
<odoo>
    <record/>
</odoo>
```

File: odoo/addons/openupgrade_records/models/comparison_config.py (hunk 7 above)
Comment (StefanRijnhart) [reply odoo/addons/openupgrade_records/models/comparison_config.py]: If you mean that translators are then confronted with %s codes in their strings, they should be used to that by now ;-)

File: scripts/compare_noupdate_xml_records.py (hunk 8 above)
Comment (duanyp1991) [reply scripts/compare_noupdate_xml_records.py]: This is when we are preparing the output.

The case u mentioned has already been considered, pls check: `process_record_node`. :)

Comment (StefanRijnhart) [reply scripts/compare_noupdate_xml_records.py]: Oh, but I was talking about the output. You are still producing files with a data node, aren't you?

Comment (duanyp1991) [reply scripts/compare_noupdate_xml_records.py]: yes that's right.

Comment (StefanRijnhart) [reply scripts/compare_noupdate_xml_records.py]: So, can you please change that?

**Comment by pedrobaeza** [comment]
This has been auto-closed trying to push latest changes. I'll open another PR.

**Comment by pedrobaeza** [comment]
Superseeded by #1008

**Pull request closed without merging**